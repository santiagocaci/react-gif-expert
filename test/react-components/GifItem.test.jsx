const { render, screen } = require('@testing-library/react');
const { GifItem } = require('../../src/react-components');

describe('pruebas en <GifItem />', () => {
  const title = 'saitama';
  const url = 'https://onepunch.com/saitama.jpg';

  test('debe de hacer match con el snapshot', () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test('debe de mostrar la imagen con el URL y el ALT indicado', () => {
    render(<GifItem title={title} url={url} />);
    // expect(screen.getByRole('img').src).toBe(url);
    // expect(screen.getByRole('img').alt).toBe(title);
    const { src, alt } = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('debe de mostrar el titulo en el componente', () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
