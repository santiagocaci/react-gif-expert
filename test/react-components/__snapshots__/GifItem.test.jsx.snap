// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifItem /> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="saitama"
      src="https://onepunch.com/saitama.jpg"
    />
    <p>
      saitama
    </p>
  </div>
</div>
`;
